multicluster/service-mirror/cluster_watcher.go

```diff
@@ -1031,6 +1031,7 @@ func (rcsw *RemoteClusterServiceWatcher) repairEndpoints(ctx context.Context) er
 	mirrorServices, err := rcsw.getMirrorServices()
 	if err != nil {
 		rcsw.log.Errorf("Failed to list mirror services: %s", err)
+		return err
```

Comment:
Where does this error get handled? Is it logged? We generally want to _either_ log _or_ return an error. I suspect we want to replace the log here with a fmt.Errorf to wrap the error type:

```suggestion
		return fmt.Errorf("failed to list mirror services: %w", err)
```

There are a few other error conditions in this function that have dubious handling:

* Directly preceding this: https://github.com/linkerd/linkerd2/blob/5ec5572ed8fc20bc2a2cd677a8259dc10b3625b3/multicluster/service-mirror/cluster_watcher.go#L1025-L1028 -- why is it safe to continue here?
* In this case, we will log and continue to use the uninitialized _endpoints_: https://github.com/linkerd/linkerd2/blob/5ec5572ed8fc20bc2a2cd677a8259dc10b3625b3/multicluster/service-mirror/cluster_watcher.go#L1058-L1063

Reply:
I believe that the operation of `createOrUpdateGatewayEndpoints` is independent of repairing the mirror services.  If the `createOrUpdateGatewayEndpoints` call fails, it should still be safe to attempt to repair the mirror services.